### ISSUE TITLE
Error in bundle exec rake db:create

### ISSUE THREAD

**Issue by caduguedess** [openstreetmap/openstreetmap-website#2292]
I'm trying to install OSM-Website in a Ubuntu Bash in Windows 10 and when I run the command:
`bundle exec rake db:create`
It gives me error:
![dbcreate](https://user-images.githubusercontent.com/51020710/60390993-d8003e80-9aba-11e9-950c-37398b64a5a4.png)
There is also a txt file uploaded for the full error with the --trace command.
[dbcreate.txt](https://github.com/openstreetmap/openstreetmap-website/files/3342427/dbcreate.txt)
Thanks for your time.

**Comment by tomhughes** [comment]
Your postgres server isn't running so you will need to consult the poistgres documentation on how to fix that.

You environment is not something that any of us have any experience in I'm afraid - we know Ubuntu but not any special issues with running it in the linux subsystem on Windows.

Normally installing postgres would have created and started an instance but it looks like that hasn't happened for you - you can try `sudo systemctl start postgresql` but if that doesn't help then I really don't know and you may need to start consulting postgres resources about whether there are any known issues running in your slightly unusual environment.

**Comment by caduguedess** [comment]
I just installed Ubuntu 18.04 in my computer, erase the Windows 10, so I will try to install in the most ideal way.
I noticed that when I installed postgre with the following code:
`sudo apt-get install postgresql postgresql-contrib libpq-dev postgresql-server-dev-all`
I think something got wrong, because I got a red message in the Terminal.
![postgre](https://user-images.githubusercontent.com/51020710/60510870-e7dc7600-9ca6-11e9-9936-dbb1c28c33c8.png)
Maybe I have to fix it before continue installing the other parts?

**Comment by tomhughes** [comment]
I honest can't remember if that is normal or not but I wouldn't worry about it unless you encounter other problems.

**Comment by gravitystorm** [comment]
I think it's fine, but you might need to run `sudo service postgresql start`.

**Comment by tomhughes** [comment]
Oh yes I sure that (or, to be modern `sudo systemctl start postgresql`) is the most that will be needed but I suspect that was probably already done and that message is just an implementation detail of it reporting status after creating the database and before starting it.

**Comment by caduguedess** [comment]
> Oh yes I sure that (or, to be modern `sudo systemctl start postgresql`) is the most that will be needed but I suspect that was probably already done and that message is just an implementation detail of it reporting status after creating the database and before starting it.

I did this command and then `service postgresql status` to check if its running, it should show active (Running) but is showing active (Exited)
![postgre](https://user-images.githubusercontent.com/51020710/60514730-74d7fd00-9cb0-11e9-9389-a2400db90b8b.png)

**Comment by dieterdreist** [comment]
try to login:
psql -U postgres
you might get an error anyway, but if it is because of a missing database,
you have the service running, if it is something like "could not connect to
server" it likely means the service is not running.

Cheers,
Martin

**Comment by caduguedess** [comment]
> try to login: psql -U postgres you might get an error anyway, but if it is because of a missing database, you have the service running, if it is something like "could not connect to server" it likely means the service is not running. Cheers, Martin

it gives `psql: FATAL:  Peer authentication failed for user "postgres"`

**Comment by tomhughes** [comment]
Please ignore @dieterdreist as that is terrible advice, though you have proved that the database is running, or you wouldn't have got that error.

**Comment by caduguedess** [comment]
Now in Ubuntu 18.04 it gives a similar error
`carlos@carlos-Inspiron-7773:~/openstreetmap-website$ bundle exec rake db:create --trace
** Invoke db:create (first_time)
** Invoke db:load_config (first_time)
** Invoke environment (first_time)
** Execute environment
pngcrush worker: `pngcrush` not found; please provide proper binary or disable this worker (--no-pngcrush argument or `:pngcrush => false` through options)
advpng worker: `advpng` not found; please provide proper binary or disable this worker (--no-advpng argument or `:advpng => false` through options)
optipng worker: `optipng` not found; please provide proper binary or disable this worker (--no-optipng argument or `:optipng => false` through options)
pngquant worker: `pngquant` not found; please provide proper binary or disable this worker (--no-pngquant argument or `:pngquant => false` through options)
jhead worker: `jhead` not found, `jpegtran` not found; please provide proper binary or disable this worker (--no-jhead argument or `:jhead => false` through options)
jpegoptim worker: `jpegoptim` not found; please provide proper binary or disable this worker (--no-jpegoptim argument or `:jpegoptim => false` through options)
jpegtran worker: `jpegtran` not found; please provide proper binary or disable this worker (--no-jpegtran argument or `:jpegtran => false` through options)
gifsicle worker: `gifsicle` not found; please provide proper binary or disable this worker (--no-gifsicle argument or `:gifsicle => false` through options)
svgo worker: `svgo` not found; please provide proper binary or disable this worker (--no-svgo argument or `:svgo => false` through options)
** Execute db:load_config
** Execute db:create
FATAL:  role "carlos" does not exist
Couldn't create 'openstreetmap' database. Please check your configuration.
rake aborted!
ActiveRecord::NoDatabaseError: FATAL:  role "carlos" does not exist
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:696:in `rescue in connect'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:691:in `connect'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:223:in `initialize'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:48:in `new'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:48:in `postgresql_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:811:in `new_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:855:in `checkout_new_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:834:in `try_to_checkout_new_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:795:in `acquire_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:523:in `checkout'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:382:in `connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:1014:in `retrieve_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_handling.rb:118:in `retrieve_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_handling.rb:90:in `connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/postgresql_database_tasks.rb:12:in `connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/postgresql_database_tasks.rb:21:in `create'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:119:in `create'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:139:in `block in create_current'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:316:in `block in each_current_configuration'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:313:in `each'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:313:in `each_current_configuration'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:138:in `create_current'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/railties/databases.rake:29:in `block (2 levels) in <top (required)>'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `block in execute'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `each'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `execute'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:214:in `block in invoke_with_call_chain'
/usr/lib/ruby/2.5.0/monitor.rb:226:in `mon_synchronize'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:194:in `invoke_with_call_chain'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:183:in `invoke'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:160:in `invoke_task'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `each'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `block in top_level'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:125:in `run_with_threads'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:110:in `top_level'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:83:in `block in run'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:186:in `standard_exception_handling'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:80:in `run'
/var/lib/gems/2.5.0/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'

Caused by:
PG::ConnectionBad: FATAL:  role "carlos" does not exist
/var/lib/gems/2.5.0/gems/pg-1.1.4/lib/pg.rb:56:in `initialize'
/var/lib/gems/2.5.0/gems/pg-1.1.4/lib/pg.rb:56:in `new'
/var/lib/gems/2.5.0/gems/pg-1.1.4/lib/pg.rb:56:in `connect'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:692:in `connect'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:223:in `initialize'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:48:in `new'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/postgresql_adapter.rb:48:in `postgresql_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:811:in `new_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:855:in `checkout_new_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:834:in `try_to_checkout_new_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:795:in `acquire_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:523:in `checkout'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:382:in `connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:1014:in `retrieve_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_handling.rb:118:in `retrieve_connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/connection_handling.rb:90:in `connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/postgresql_database_tasks.rb:12:in `connection'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/postgresql_database_tasks.rb:21:in `create'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:119:in `create'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:139:in `block in create_current'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:316:in `block in each_current_configuration'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:313:in `each'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:313:in `each_current_configuration'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:138:in `create_current'
/var/lib/gems/2.5.0/gems/activerecord-5.2.3/lib/active_record/railties/databases.rake:29:in `block (2 levels) in <top (required)>'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `block in execute'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `each'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:273:in `execute'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:214:in `block in invoke_with_call_chain'
/usr/lib/ruby/2.5.0/monitor.rb:226:in `mon_synchronize'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:194:in `invoke_with_call_chain'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/task.rb:183:in `invoke'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:160:in `invoke_task'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `each'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:116:in `block in top_level'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:125:in `run_with_threads'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:110:in `top_level'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:83:in `block in run'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:186:in `standard_exception_handling'
/var/lib/gems/2.5.0/gems/rake-12.3.2/lib/rake/application.rb:80:in `run'
/var/lib/gems/2.5.0/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
/usr/local/bin/rake:23:in `load'
/usr/local/bin/rake:23:in `<main>'
Tasks: TOP => db:create
carlos@carlos-Inspiron-7773:~/openstreetmap-website$ 
`

**Comment by caduguedess** [comment]
Now in Ubuntu 18.04 it gives a similar error.
![rake1](https://user-images.githubusercontent.com/51020710/60620018-77ba1700-9db0-11e9-9f56-b51108bdc4a9.png)
![rake2](https://user-images.githubusercontent.com/51020710/60620033-7d176180-9db0-11e9-8173-02af0b1e3c24.png)
![rake3](https://user-images.githubusercontent.com/51020710/60620041-7ee12500-9db0-11e9-91c8-1696cda5c215.png)

**Comment by caduguedess** [comment]
> I think it's fine, but you might need to run `sudo service postgresql start`.

didn't work

**Comment by tomhughes** [comment]
No, postgres is running, you just haven't created a postgres user for your OS account - see the section headed `PostgreSQL account setup` in the instructions.

**Comment by caduguedess** [comment]
I just noticed I was doing that command outside the folder `openstreetmap website`
Now it seems like worked, but with some weird "not found" messages.
T
![notf](https://user-images.githubusercontent.com/51020710/60620765-72f66280-9db2-11e9-8e05-d08d410d122e.png)
hanks

**Comment by tomhughes** [comment]
That's fine - all those programs are optional.

**Comment by caduguedess** [comment]
I followed the instructions and got stuck at this.
When I run the `bundle exec rake test:db` I get a lot of errors.
https://gist.github.com/caduguedess/4cd6d9205f741e105722d53451bb46c2

**Comment by caduguedess** [comment]
I did `sudo apt-get install phantomjs` and now `bundle exec rake test:db` runs a long test but with scaring errors.
![error](https://user-images.githubusercontent.com/51020710/60679470-10f33700-9e5e-11e9-81d4-c3c068eff743.png)

**Comment by caduguedess** [comment]
And this is my localhost:3000 after starting the server
![Screenshot from 2019-07-04 13-20-18](https://user-images.githubusercontent.com/51020710/60679604-94148d00-9e5e-11e9-9f7b-4abb848fc1e7.png)

**Comment by gravitystorm** [comment]
This is a bug in the way that the codebase works right now. You need to create a blank file in the config directory called settings.local.yml, for example by running

```
$ touch config/settings.local.yml
```
See https://github.com/openstreetmap/openstreetmap-website/issues/2185 for more information.

**Comment by caduguedess** [comment]
I reduced the errors to only 3.
The first and second errors are about duplicated accounts created, and the last is an failure as shown in the link below
https://gist.github.com/caduguedess/5018e16746c538628e8060c135c42a40
Is it something I need to worry?
The localhost:3000 is already showing the map

**Comment by gravitystorm** [comment]
Please update your source code. It looks like you are hitting a problem with one of the translation files that was fixed in https://github.com/openstreetmap/openstreetmap-website/commit/ea2c790a42cbb4ed6582646b8f45af5050ba472c

**Comment by tomhughes** [comment]
No it's come back again - will fix it again shortly...

**Comment by tomhughes** [comment]
Oh no it has actually been fixed on TW now.

**Comment by caduguedess** [comment]
> Please update your source code. It looks like you are hitting a problem with one of the translation files that was fixed in [ea2c790](https://github.com/openstreetmap/openstreetmap-website/commit/ea2c790a42cbb4ed6582646b8f45af5050ba472c)

I do this just by running `git clone --depth=1 https://github.com/openstreetmap/openstreetmap-website.git` ?

**Comment by caduguedess** [comment]
I tried by doing `sudo apt-get update` and now got more errors, they are related to unitialized constants.
![constants](https://user-images.githubusercontent.com/51020710/60730933-a1904c80-9f1c-11e9-8419-85e2708c36c2.png)

**Comment by gravitystorm** [comment]
> I tried by doing sudo apt-get update 

That doesn't update the codebase, just the underlying system libraries. 

Updating your copy of the codebase using git can be done in a number of ways, depending on whether you've made changes to the code on your system. If you haven't then `git pull` is usually sufficient. But there are many tutorials on how to use git available online that will do a better job of explaining it than I will!

**Comment by caduguedess** [comment]
I just edited the file manually and now it gives no error.

**Comment by caduguedess** [comment]
I'm not receiving the confirmation email of sign up.
In  #2138 it seems to take 4 hours, but is it supposed to take that long?

**Comment by caduguedess** [comment]
In https://help.openstreetmap.org/questions/8526/how-to-edit-my-private-osm-server-map an user said that its needed a tile server for using it.
Couldn't the data be stored in the same PostGIS database that were used to create the OSM Website?

**Comment by tomhughes** [comment]
Firstly the web site doesn't use a postigs database - it uses a postgres database without the postgis extension.

Secondly in a literal sense it could be stored in the same database but as the schema is completely different the data would all be duplicated - the rendering data is held as postgis geometries while the main database is a database of our of design that preserves topology (how roads etc interconnect) which is not possible when using geometries.

**Comment by caduguedess** [comment]
Thanks.
No problem, I will follow https://switch2osm.org/manually-building-a-tile-server-18-04-lts/ to setup a tile server then.
Btw, is the IRC channel #osm-dev still working?
I can't connect to it.
